server-spi/src/main/java/org/keycloak/models/RealmProvider.java

```diff
@@ -147,14 +177,14 @@ default ClientModel addClient(RealmModel realm, String clientId) {
     /**
      * @deprecated Use the corresponding method from {@link ClientProvider}. */
     @Override
-    default List<ClientModel> getClients(RealmModel realm) {
-        return this.getClients(realm, null, null);
+    default Stream<ClientModel> getClientsStream(RealmModel realm) {
+        return this.getClientsStream(realm, null, null);
     }
 
     /**
      * @deprecated Use the corresponding method from {@link ClientProvider}. */
     @Override
-    public List<ClientModel> getClients(RealmModel realm, Integer firstResult, Integer maxResults);
+    public Stream<ClientModel> getClientsStream(RealmModel realm, Integer firstResult, Integer maxResults);
```

Comment:
In case I revert this I have to implement the method in `JpaRealmProvider` and `RealmCacheSession` or the method has to be removed from `RealmProvider`.

Reply:
I proceeded with removing the method from RealmProvider.